BraveWallet/Crypto/Stores/KeyringStore.swift

```diff
@@ -96,10 +96,15 @@ public class KeyringStore: ObservableObject {
   }
   
   private let keyringService: BraveWalletKeyringService
+  private let walletService: BraveWalletBraveWalletService // for wallet reset
```

Comment:
Remove this and `reset` from `KeyringStore` and use wallet service directly in settings instead. Then use the observer to handle the logic from reset (`isOnboardingVisible`/`Self.resetKeychainStoredPassword`)